### ISSUE TITLE
NUnit2021 needs to be aware of PropertiesComparer

### ISSUE THREAD

**Issue by manfred-brands** [nunit/nunit.analyzers#877]
The PropertiesComparer has an overload to only compare common properties. In those cases, the type is not relevant:

```csharp
Assert.That(instanceS, Is.EqualTo(instanceC).UsingPropertiesComparer(
    c => c.CompareOnlyCommonProperties()));
```

The PropertiesComparer also allows mapping properties and types:

```csharp
Assert.That(usPerson, Is.EqualTo(person).UsingPropertiesComparer(
    c => c.Map<USPerson>(x => x.Address, y => y.USAddress)
            .Map<Address, USAddress>(x => x.AreaCode, y => y.ZipCode)
            .Map<Address>(x => x.Country, "U.S.A.")));
```

**Comment by manfred-brands** [comment]
@mikkelbu Yes it is a duplicate. I tried upgrading NUnit to use the latest analyzer and noticed this and without checking added a new ticket.

`UsingPropertiesComparer` on its own requires the same types. Only when a modified is used it allows different types. In the latter case the rule should not fire.
